### ISSUE TITLE
InputEventFetcher process starves image

### ISSUE THREAD

**Issue by smarr** [smarr/RoarVM#1]
The input poll process starves the image.
The eventBuffer in InputEventFetcher>>eventLoop is not filled in correctly, and the loop will end go on infinitely.

The problem is either the primitive or something Pharo specific. This problem is not present in Squeak 4.1 and MVC images

The current work around is:

```
!InputEventFetcher methodsFor: 'events' stamp: 'StefanMarr 10/31/2010 21:45' prior: 47875547!
eventLoop
    "Fetch pending raw events from the VM.
     This method is run at high priority."


    [true] whileTrue: [
        | window eventBuffer type |
        self waitForInput.
        [eventBuffer := Array new: 8.
        self primGetNextEvent: eventBuffer.
        type := eventBuffer at: 1.
        ((type = EventTypeNone) or: [type isNil])]
            whileFalse: [
                "Patch up the window index in case we don't get one"
                window := eventBuffer at: 8.
                (window isNil
                    or: [window isZero])
                    ifTrue: [eventBuffer at: 8 put: 1]. 

                self signalEvent: eventBuffer]].! !
```

**Comment by smarr** [comment]
A better fix, which is equivalent to what all other forks do is as follows:

```
!InputEventFetcher methodsFor: 'private' stamp: 'StefanMarr 11/4/2010 00:01' prior: 22276785!
primGetNextEvent: array
    "Store the next OS event available into the provided array.
    Essential."
    <primitive: 94>
    array at: 1 put: EventTypeNone. "If there is no event, the primitive will fail."
    ^nil! !
```

**Comment by smarr** [comment]
Fixed in Pharo 1.2
http://code.google.com/p/pharo/issues/detail?id=3209
